sdk/src/main/scala/com/horizen/storage/SidechainStateStorage.scala

```
@@ -304,4 +304,42 @@ class SidechainStateStorage(storage: Storage, sidechainBoxesCompanion: Sidechain
 
   def isEmpty: Boolean = storage.isEmpty
 
+  def getIterator: StorageIterator = storage.getIterator
+
+  /**
+   * This function restores the unspent boxes that come from a ceased sidechain by saving
+   * them into the SidechainStateStorage
+   *
+   * @param backupStorage: storage containing the boxes saved from the ceased sidechain
+   */
+  def restoreBackup(backupStorageIterator: StorageIterator, lastVersion: Array[Byte]): Unit = {
+    backupStorageIterator.seekToFirst()
+    val removeList = new JArrayList[ByteArrayWrapper]()
+    val updateList = new JArrayList[JPair[ByteArrayWrapper,ByteArrayWrapper]]()
+    val lastVersionWrapper = new ByteArrayWrapper(lastVersion)
+
+    while(backupStorageIterator.hasNext) {
+      val entry = backupStorageIterator.next()
+      val box: Try[SCB] = sidechainBoxesCompanion.parseBytesTry(entry.getValue)
+
+      if (box.isSuccess) {
+        val currBox: SCB = box.get
+        if (!currBox.isInstanceOf[CoinsBox[_ <: PublicKey25519Proposition]]) {
```

`else throw exception`. we have to forbid coinbox backup operation.
I think, would be nice to even skip coin boxes while looping through the iterator on backup phase.

---

Good point, i added a skip for coin boxes inside the backup phase.
At this point do you think is better to throw an exception that interrupts the restore, instead of ignore the specific box?

---

I think, it is better to interrupt and highlight the error to the user.
Otherwise, the result can be different to what user expected. And he will not understand why.